### ISSUE TITLE
Trouble building Forms Project with Xamarin.Mac - Microsoft.NuGet.targets error

### ISSUE THREAD

**Issue by chamons** [xamarin/Xamarin.Forms#3316]
### Description

Originally reported via twitter: https://twitter.com/MRumpler/status/1018857084346097664

Microsoft.NuGet.targets appears to be failing to handle Xamarin.Mac Modern correctly, which appears to be getting pulled in via Forms.

### Steps to Reproduce

1. Open attached project
2. Build

### Expected Behavior

Successful build or better error.

### Actual Behavior

```/Library/Frameworks/Mono.framework/Versions/5.10.1/lib/mono/xbuild/Microsoft/NuGet/Microsoft.NuGet.targets(184,5): error : Your project is not referencing the "Xamarin.Mac,Version=v2.0" framework. Add a reference to "Xamarin.Mac,Version=v2.0" in the "frameworks" section of your project.json, and then re-run NuGet restore.```

### Basic Information

- Version with issue: https://gist.github.com/chamons/78af2a796375b7fb7bfdb77ca1fe26d8

### Reproduction Link

[MR.Gestures.zip](https://github.com/xamarin/Xamarin.Forms/files/2199073/MR.Gestures.zip)

**Comment by MichaelRumpler** [comment]
I tried several things yesterday like switching from stable to beta and alpha channels and then back to stable. Nothing worked.
Today I installed Xamarin.Forms 3.1.0.637273 accidentally and all of a sudden it compiled. Also when I removed the package and installed version 3.0.0.482510 again (as I don't want to depend on 3.1).

I don't know, which of those actions fixed it or if they were needed all together, but now it builds for me. So this issue can be closed. Thanks for your efforts!
